runtime/commonMain/src/kotlinx/serialization/Encoding.kt

```
@@ -322,6 +330,12 @@ public interface CompositeEncoder {
      */
     public fun encodeByteElement(descriptor: SerialDescriptor, index: Int, value: Byte)
 
+    /**
+     * Encodes the bytes [value] associated with an element at the given [index] in [serial descriptor][descriptor].
+     * The element at the given [index] should be a [ByteArray].
+     */
+    public fun encodeByteStringElement(descriptor: SerialDescriptor, index: Int, value: ByteString) {}
```

Rather than an empty function, can we throw `UnsupportedOperation` (or similar) except for CBOR encoding?